### PR TITLE
u001-font: init at unstable-2016-08-01

### PR DESCRIPTION
**Pull request by toastal**
###### Description of changes

Added a Univers-like font that is somewhat lost to time and with a weird, non-free but free-adjacent LICENSE from an earlier time of URW++ and. The Aladdin license is very much against you selling it, but requires you to distribute it--more recently their fonts are AGPL or GPL, but this one doesn't seem to be re-licensed and lives on with its old license.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: lib/licenses.nix [hunk 1]
```diff
@@ -55,6 +55,12 @@ in mkLicense lset) ({
     fullName = "GNU Affero General Public License v3.0 or later";
   };
 
+  aladdin = {
```
Comment (azahi) [inline lib/licenses.nix:58]: Maybe it's better to name it `afpl` as the Wikipedia [page](https://en.wikipedia.org/wiki/Aladdin_Free_Public_License) suggests?

File: pkgs/data/fonts/u001/default.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{ lib, stdenv, fetchurl, unzip }:
+
+stdenv.mkDerivation rec {
+  name = "U001";
+  pname = "u001";
```
Comment (azahi) [inline pkgs/data/fonts/u001/default.nix:5]: IMO this is way too confusing, coupled with the fact that the package is essentially will be called `u001-font` in `all-packages.nix`. Maybe having a unified name would be better?

File: pkgs/data/fonts/u001/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{ lib, stdenv, fetchurl, unzip }:
+
+stdenv.mkDerivation rec {
```
Comment (azahi) [inline pkgs/data/fonts/u001/default.nix:3]: Not sure about this, but wouldn't it be better to use `stdenvNoCC`?
```suggestion
stdenvNoCC.mkDerivation rec {
```

File: pkgs/data/fonts/u001/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    cp u001con-ita.ttf $TTF/U001Condensed-Italic.ttf
+    cp u001con-bol.ttf $TTF/U001Condensed-Bold.ttf
+    cp u001con-bolita.ttf $TTF/U001Condensed-BoldItalic.ttf
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A Univers-like typeface that comes with GhostPDL made by URW++";
+    license = licenses.aladdin;
```
Comment (azahi) [inline pkgs/data/fonts/u001/default.nix:48]: Where did you find what license the font has? I can only see this on their [homepage](https://www.cufonfonts.com/font/u001):

> License
> (URW)++,Copyright 2000 by (URW)++ Design & Development

File: lib/licenses.nix (hunk 1 above)
Comment (toastal) [reply lib/licenses.nix:58]: I thought about that but https://spdx.org/licenses/ shows that the SPDX identifier as "Aladdin" which made more sense to match.

File: pkgs/data/fonts/u001/default.nix (hunk 2 above)
Comment (toastal) [reply pkgs/data/fonts/u001/default.nix:5]: I'm not really sure what the best option here is. I disagree with not adding `-ttf` or `-font` as these will run into naming collisions fast and fonts are a different kind of software. I can ask around for suggestions as well on what to do here.

File: pkgs/data/fonts/u001/default.nix (hunk 4 above)
Comment (toastal) [reply pkgs/data/fonts/u001/default.nix:48]: https://fontlibrary.org/en/font/u001 

Shows Aladdin but is only the Condensed fonts (it's also the page referenced in the Wikipedia article)

https://en.wikipedia.org/wiki/Univers#URW_Classic_Sans

The Wikipedia section mentions it's a part of GhostPDL.

https://en.wikipedia.org/wiki/Ghostscript#Free_fonts

Shows U001 under the list of fonts under AFPL

File: pkgs/data/fonts/u001/default.nix (hunk 3 above)
Comment (toastal) [reply pkgs/data/fonts/u001/default.nix]: I did not know about this. Updated.

File: pkgs/data/fonts/u001/default.nix (hunk 4 above)
Comment (toastal) [reply pkgs/data/fonts/u001/default.nix:48]: I omitted a `homepage` because it doesn't seem to really have a "home" anymore and has just been hosted in some places (some of which lacking the correct license line, though the font from cufonts does support 4 styles in both regular and codensed).

File: pkgs/data/fonts/u001/default.nix (hunk 2 above)
Comment (toastal) [reply pkgs/data/fonts/u001/default.nix:5]: Though now that I think about it, this seems more like a `nixpkgs` org issue--that fonts should probably be @ `font.XXX` just like `haskellPackages.XXX`.

File: pkgs/data/fonts/u001/default.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    cp u001con-reg.ttf $TTF/U001Condensed-Regular.ttf
+    cp u001con-ita.ttf $TTF/U001Condensed-Italic.ttf
+    cp u001con-bol.ttf $TTF/U001Condensed-Bold.ttf
+    cp u001con-bolita.ttf $TTF/U001Condensed-BoldItalic.ttf
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A Univers-like typeface that comes with GhostPDL made by URW++";
```
[the remark this answers is not in the document]
Comment (toastal) [reply pkgs/data/fonts/u001/default.nix:47]: See other comments about why I omitted the homepage... unless you have a better idea, I don't believe this is a very good homepage considering that the Aladdin license info is missing and it's sort of an anyone-can-upload place.

File: lib/licenses.nix (hunk 1 above)
Comment (AndersonTorres) [inline lib/licenses.nix:58]: Put this modification in a separate commit.

File: pkgs/data/fonts/u001/default.nix (hunk 5 above)
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/data/fonts/u001/default.nix:47]: Maybe https://fontlibrary.org/en/font/u001  should serve?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/data/fonts/u001/default.nix (hunk 5 above)
[the remark this answers is not in the document]
Comment (toastal) [reply pkgs/data/fonts/u001/default.nix:47]: You know, you are 100% correct. When I looked earlier, I didn't see the non-condensed fonts so I thought they were missing, but I see it now. I will change this right away.

[the remark this answers is not in the document]
Comment (toastal) [reply pkgs/data/fonts/u001/default.nix:40]: Done.

File: pkgs/data/fonts/u001/default.nix (hunk 4 above)
Comment (toastal) [reply pkgs/data/fonts/u001/default.nix:42]: Moved to the https://fontlibrary.org/en/font/u001 option

File: lib/licenses.nix (hunk 1 above)
Comment (toastal) [reply lib/licenses.nix:58]: Added a separate commit before. Do you have a suggestion about the `aladdin` vs `afpl` naming?

**Review by azahi — CHANGES_REQUESTED** [review]
LGTM apart from the AFPL attribute name (although, I'm not sure what's the best name) and commit messages: they should be "**lib.licenses:** add Aladdin Free Public License" and 
"u001-font: init at unstable<b>-</b>2016-08-01" (note the dash)

**Comment by toastal** [comment]
@azahi good catch. Pushed commit messages.

**Review by azahi — APPROVED** [review]

File: pkgs/data/fonts/u001/default.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    cp u001-reg.ttf $TTF_DIR/U001-Regular.ttf
+    cp u001-ita.ttf $TTF_DIR/U001-Italic.ttf
+    cp u001-bol.ttf $TTF_DIR/U001-Bold.ttf
+    cp u001-bolita.ttf $TTF_DIR/U001-BoldItalic.ttf
+    cp u001con-reg.ttf $TTF_DIR/U001Condensed-Regular.ttf
+    cp u001con-ita.ttf $TTF_DIR/U001Condensed-Italic.ttf
+    cp u001con-bol.ttf $TTF_DIR/U001Condensed-Bold.ttf
+    cp u001con-bolita.ttf $TTF_DIR/U001Condensed-BoldItalic.ttf
+
+    cp Copying.AFPL.txt $DOC_DIR
```
Comment (AndersonTorres) [inline pkgs/data/fonts/u001/default.nix:33]: There is no necessity of including license files.
The `meta.license` fulfills this role.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: lib/licenses.nix (hunk 1 above)
Comment (AndersonTorres) [reply lib/licenses.nix:58]: SPDX is my source of truth on this case. And they call it Aladdin!

https://spdx.org/licenses/Aladdin.html

File: pkgs/data/fonts/u001/default.nix (hunk 6 above)
Comment (AndersonTorres) [reply pkgs/data/fonts/u001/default.nix:33]: IMHO the only justifiable exceptions should be

1. the build script shipped with the source code installs the license file; and/or
2. the license file is used by the program in runtime (think on an ABOUT button)

File: pkgs/data/fonts/u001/default.nix [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+  src = fetchzip {
+    url = "https://fontlibrary.org/assets/downloads/u001/3ea00b3c0c8fa6ce4373e5766fafd651/u001.zip";
+    sha256 = "sha256-7H32pfr0g68XP5B48VUY99e6fbd7rhH6fEnCKNXWEkU=";
+    stripRoot = false;
+  };
+
+  dontBuild = true;
+
+  # We’ll adjust the nonstandard naming convention here
```
Comment (AndersonTorres) [inline pkgs/data/fonts/u001/default.nix:15]:
```suggestion
```

File: pkgs/data/fonts/u001/default.nix [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    TTF_DIR=$out/share/fonts/truetype
+    DOC_DIR=$out/share/doc
+
+    mkdir -p $TTF_DIR $DOC_DIR
+
```
Comment (AndersonTorres) [inline pkgs/data/fonts/u001/default.nix:23]:
```suggestion
  # Adjust the nonstandard naming conventions
```

File: pkgs/data/fonts/u001/default.nix (hunk 7 above)
Comment (AndersonTorres) [reply pkgs/data/fonts/u001/default.nix:15]: Better to move the comment near to the code it refers

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
